plugins/org.jboss.tools.jst.angularjs/src/org/jboss/tools/jst/angularjs/internal/ExpressionsProposalProcessor.java

```diff
@@ -0,0 +1,163 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.jst.angularjs.internal;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.jboss.tools.common.text.TextProposal;
+import org.jboss.tools.jst.angularjs.AngularJsPlugin;
+import org.jboss.tools.jst.web.kb.IPageContext;
+import org.jboss.tools.jst.web.kb.IProposalProcessor;
+import org.jboss.tools.jst.web.kb.KbQuery;
+import org.jboss.tools.jst.web.kb.internal.BrowserDataProviderManager;
+
+/**
+ * @author Alexey Kazakoc
+ */
+public class ExpressionsProposalProcessor implements IProposalProcessor {
+
+	private ImageDescriptor image;
+
+	/* (non-Javadoc)
+	 * @see org.jboss.tools.jst.web.kb.IProposalProcessor#getProposals(org.jboss.tools.jst.web.kb.KbQuery, org.jboss.tools.jst.web.kb.IPageContext)
+	 */
+	@Override
+	public TextProposal[] getProposals(KbQuery query, IPageContext context) {
+		List<TextProposal> proposals = new ArrayList<TextProposal>();
+		String qValue = query.getValue();
+		int startEl = qValue.lastIndexOf("{{");
+		if(startEl<0) {
+			qValue = query.getRegionValuePrefix();
+			if(qValue!=null) {
+				startEl = qValue.lastIndexOf("{{");
+			}
+		}
+		if(startEl>-1) {
+			String value = qValue.substring(startEl + 2);
+			if(!value.contains("\n") && !value.contains("}") && (new AngularJSRecognizer()).isUsed(context)) {
+				// Create a JS
+				String js = generateJs(value, context);
+				Collection<Object> results = BrowserDataProviderManager.getInstance().evaluate(js, context);
+				// Convert the data to a list of proposals
+				if(!results.isEmpty()) {
+					if(image == null) {
+						image = ImageDescriptor.createFromFile(AngularJsPlugin.class, "angular.png");
```

Comment:
We don't create/use/have an actual Image. This is an image descriptor which we provide to the Eclipse content assist and it will do the rest of the work to create and probably cache the Image(s).
And we reuse the same image descriptor in angularjs content assist every time you use it.